mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    is_lot_number_propagated = fields.Boolean(
+        related="bom_id.lot_number_propagation",
```

Comment:
After testing, I feel like this field should not be related to the BOM, but set through `action_confirm`. So that the MOs that were confirmed won't be affected by changes to the BOM (as is the case with components).

Reply:
Stored on `action_confirm` + added an `onchange` on `bom_id` to have the value even if the MO is not yet validated